gax-httpjson/src/test/java/com/google/api/gax/httpjson/HttpJsonDirectCallableTest.java

```diff
@@ -151,6 +151,80 @@ public void testSuccessfulUnaryResponse() throws ExecutionException, Interrupted
     assertThat(headerValue).isEqualTo("headerValue");
   }
 
+  /**
+   * This test is for a Unary Call where server mistakenly sends multiple responses back The
+   * expectation for this MOCK_SERVICE is to return what was sent into the request i.e.
+   * callable.futureCall(x) -> x
+   *
+   * <p>For a Unary Call, gax will return only the first (and hopefully only) response back.
+   *
+   * @throws InterruptedException
+   * @throws ExecutionException
+   */
+  @Test
+  public void testSuccessfulMultipleResponsesForUnaryCall()
+      throws InterruptedException, ExecutionException {
+    HttpJsonDirectCallable<Field, Field> callable =
+        new HttpJsonDirectCallable<>(FAKE_METHOD_DESCRIPTOR);
+
+    HttpJsonCallContext callContext = HttpJsonCallContext.createDefault().withChannel(channel);
+
+    Field request;
+    Field expectedResponse;
+    request = expectedResponse = createTestMessage(2);
+    Field otherResponse = createTestMessage(10);
+    MOCK_SERVICE.addResponse(expectedResponse);
+    MOCK_SERVICE.addResponse(otherResponse);
+    MOCK_SERVICE.addResponse(otherResponse);
+
+    Field actualResponse = callable.futureCall(request, callContext).get();
+    assertThat(actualResponse).isEqualTo(expectedResponse);
+    assertThat(MOCK_SERVICE.getRequestPaths().size()).isEqualTo(1);
+    String headerValue = MOCK_SERVICE.getRequestHeaders().get("header-key").iterator().next();
+    assertThat(headerValue).isEqualTo("headerValue");
+  }
+
+  /**
+   * This test is for a Unary Call where server mistakenly sends multiple responses back The
+   * expectation for this MOCK_SERVICE is to return what was sent into the request i.e.
+   * callable.futureCall(x) -> x
+   *
+   * <p>For a Unary Call, gax will return only the first (and hopefully only) response back.
+   *
+   * @throws InterruptedException
+   * @throws ExecutionException
+   */
+  @Test
+  public void testErrorMultipleResponsesForUnaryCall()
+      throws InterruptedException, ExecutionException {
+    HttpJsonDirectCallable<Field, Field> callable =
+        new HttpJsonDirectCallable<>(FAKE_METHOD_DESCRIPTOR);
+
+    HttpJsonCallContext callContext = HttpJsonCallContext.createDefault().withChannel(channel);
+
+    Field request;
+    Field expectedResponse;
+    request = expectedResponse = createTestMessage(2);
```

Comment:
I feel like 
```
    Field request = createTestMessage(2);
    Field expectedResponse = createTestMessage(2);
```
is clearer. 
Also I'm not sure if the name makes sense, if we are asserting actual response to be `randomResponse1`, then we should probably rename `randomResponse1` to `expectedResponse`.

Reply:
Sure, I can update.

I wanted the assertions below to show that it wasn't the expected response:
```
    // Gax returns the first response for Unary Call
    assertThat(actualResponse).isEqualTo(randomResponse1);
    assertThat(actualResponse).isNotEqualTo(expectedResponse);
```